include/aws/http/server.h

```diff
@@ -163,14 +171,15 @@ AWS_HTTP_API
 struct aws_http_server *aws_http_server_new(const struct aws_http_server_options *options);
 
 /**
- * Destroy server.
+ * Release the server. It will close the listening socket and all the connections existing in the server.
+ * The on_destroy_complete will be invoked when the destroy operation completes
  *
  * Note: this function should be called by either a user thread (like the main entry point, or from the event-loop the
  * server is assigned to. Otherwise a deadlock is possible. If you call this function from outside the assigned
  * event-loop, this function will block waiting on the assigned event-loop runs the close sequence in its thread.
  */
 AWS_HTTP_API
-void aws_http_server_destroy(struct aws_http_server *server);
+void aws_http_server_release(struct aws_http_server *server);
```

Comment:
Yes, the shutdown process is async now, but the aws_server_bootstrap_destroy_socket_listener() will only be called from the aws_http_server_release() function and it will only be called from the user or the server event loop, thus I think there is no problem here

Reply:
I did this in http server, I guess we can also do the same thing in IO